x-pack/plugins/ml/public/application/jobs/new_job/job_from_lens/quick_create_job.ts

```diff
@@ -330,4 +338,35 @@ export class QuickJobCreator {
 
     return mergedQueries;
   }
+
+  private async createDashboardLink(dashboard: Dashboard, datafeedConfig: estypes.MlDatafeed) {
+    if (dashboard === undefined) {
+      // embeddable may have not been in a dashboard
+      return null;
+    }
+
+    const params: DashboardAppLocatorParams = {
+      dashboardId: dashboard.id,
+      timeRange: {
+        from: '$earliest$',
+        to: '$latest$',
+        mode: 'absolute',
+      },
+      filters: getFiltersForDSLQuery(
+        datafeedConfig.query,
+        undefined,
+        undefined,
+        FilterStateStore.GLOBAL_STATE
+      ),
+    };
+    const dashboardLocator = this.share.url.locators.get('DASHBOARD_APP_LOCATOR');
+    const url = dashboardLocator ? await dashboardLocator.getUrl(params) : '';
+    return { url_name: 'Original dashboard', url_value: url };
```

Comment:
You should add `"time_range": "auto"` in here, so that the link from the anomalies table uses the time range from the anomaly. Otherwise it is defaulting to `Last 15 minutes` for me.

Reply:
This problem was being caused by the dashboards `getUrl` function encoding the url string.
Fixed in https://github.com/elastic/kibana/pull/142139/commits/53b8760d830ed0a3d79f7fd441a4de8bad284216